src/main.ts

```diff
@@ -7,15 +8,55 @@ import {
   SwaggerCustomOptions,
   SwaggerModule,
 } from '@nestjs/swagger';
+import { WinstonModule } from 'nest-winston';
+import * as winston from 'winston';
 
 import { AppModule } from './app.module';
 
 async function bootstrap() {
-  const app = await NestFactory.create(AppModule);
+  // Improved Winston Logger with environment-based transports
+  const logger = WinstonModule.createLogger({
+    level: process.env.NODE_ENV === 'production' ? 'info' : 'debug',
+    transports: [
+      new winston.transports.Console({
+        format: winston.format.combine(
+          winston.format.colorize({ all: true }),
+          winston.format.timestamp({ format: 'YYYY-MM-DD HH:mm:ss' }),
+          winston.format.printf(
+            ({ timestamp, level, message, context, stack }) => {
+              return `${timestamp} [${level}]${context ? ' [' + context + ']' : ''
+                }: ${message} ${stack ? '\n' + stack : ''}`;
+            },
+          ),
+        ),
+      }),
+      new winston.transports.DailyRotateFile({
+        filename: 'logs/application-%DATE%.log',
+        datePattern: 'YYYY-MM-DD',
+        zippedArchive: true,
+        maxSize: '20m',
+        maxFiles: '14d',
+        format: winston.format.combine(
+          winston.format.timestamp(),
+          winston.format.json(),
+        ),
+        level: 'info',
+      }),
+    ],
+
+    silent: process.env.NODE_ENV === 'test',
+  });
```

Comment:
Please, move this code to another file called winston.logger.ts.

and import like this:

```
// imports for Winston logger
import { WinstonModule } from 'nest-winston';
import { instance } from 'logger/winston.logger';
```

Reply:
Its important show  logs for diferents environments, on develop more readable and more JSON on production.

```
import { createLogger, format, transports } from "winston";

// custom log display format
const customFormat = format.printf(({timestamp, level, stack, message}) => {
    return `${timestamp} - [${level.toUpperCase().padEnd(7)}] - ${stack || message}`
})

const options = {
    file: {
        filename: 'error.log',
        level: 'error'
    },
    console: {
        level: 'silly'
    }
}

// for development environment
const devLogger = {
    format: format.combine(
        format.timestamp(),
        format.errors({stack: true}),
        customFormat
    ),
    transports: [new transports.Console(options.console)]
}

// for production environment
const prodLogger = {
    format: format.combine(
        format.timestamp(),
        format.errors({stack: true}),
        format.json()
    ),
    transports: [
        new transports.File(options.file),
        new transports.File({
            filename: 'combine.log',
            level: 'info'
        })
    ]
}

// export log instance based on the current environment
const instanceLogger = (process.env.NODE_ENV === 'production') ? prodLogger : devLogger

export const instance = createLogger(instanceLogger)

```